test/jasmine/tests/bar_test.js

```diff
@@ -2548,3 +2549,96 @@ function assertTraceField(calcData, prop, expectation) {
 
     expect(values).toBeCloseToArray(expectation, undefined, '(field ' + prop + ')');
 }
+
+describe('bar tweening', function() {
+    var gd;
+    var mock = {
+        'data': [{
+            'type': 'bar',
+            'x': ['A', 'B', 'C'],
+            'text': ['A', 'B', 'C'],
+            'textposition': 'inside',
+            'y': [24, 5, 8],
+            'error_y': {'array': [3, 2, 1]}
+        }]
+    };
+    var transitionOpts = false; // use default
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    afterEach(function() {
+        Plotly.purge(gd);
+        destroyGraphDiv();
+    });
+
+    it('for bar fill color', function(done) {
```

Comment:
These tests are :100: 

From your

- [x] length
- [x] width
- [x] text translates, rotates and scales properly
- [x] fill color
- [x] error bars
- [x] border width and color

checklist, looks like we're only missing :lock:s for `marker.line.width` and `marker.line.color`

Reply:
and could you add one test for horizontal bars to test this clause:

https://github.com/plotly/plotly.js/blob/bda810fd90be1ccb6993ec856d8ff59f007aa772/src/traces/bar/plot.js#L141

Reply:
> checklist, looks like we're only missing s for `marker.line.width` and `marker.line.color`

Added in https://github.com/plotly/plotly.js/pull/4180/commits/cc871cdb5292bf7c3e75431d3936c2229966cb37